sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -301,6 +304,31 @@ case class StringLength(child: Expression) extends UnaryExpression with ExpectsI
 }
 
 /**
+ * A function that return soundex code of the given string expression.
+ */
+case class SoundEx(child: Expression) extends UnaryExpression with ExpectsInputTypes {
+  @transient lazy val so = new Soundex
+
+  override def dataType: DataType = StringType
+
+  override def inputTypes: Seq[DataType] = Seq(StringType)
+
+  override def eval(input: InternalRow): Any = {
+    val string = child.eval(input)
+    if (string == null) {
+      null
+    } else {
+      try {
+        val str: String = string.toString
+        UTF8String.fromString(so.encode(str))
```

Comment:
@HuJiayin I was talking about this line (in the other pr #7208). Even if `str = ""` Soundex gets created (because of `so.encode`), but as @chenghao-intel mentioned, it might be no big deal.